### PR TITLE
Add keystone auth and mysql acl

### PR DESCRIPTION
**Pull request by xwisen**
branch `auth` main changes:

1.Add keystone_auth.go[check user and password with keystone]
2.Add acl_mysql.go[use mysql store relationship of multi-project and multi-user]
3.Modify server.go[offer API to user for change mysql data]

### CODE REVIEW

File: auth_server/authn/keystone_auth.go [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+		},
+	}
+
+	cpio, err := json.Marshal(cp)
+	if err != nil {
+		glog.Errorln(err)
+	}
+	req, err := http.NewRequest("POST", reqURl, bytes.NewBuffer(cpio))
+	if err != nil {
+		fmt.Println(err)
```
Comment (zhuguihua) [inline auth_server/authn/keystone_auth.go]: If error was occured, you must return from the function. Only to print the error message is not enough. Please check other places.

File: auth_server/authn/keystone_auth.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
+package authn
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
```
Comment (zhuguihua) [inline auth_server/authn/keystone_auth.go]: Please put the line below "net/http".

File: auth_server/authn/keystone_auth.go [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+	"io/ioutil"
+	"net/http"
+)
+
+//const (
+//	KEYSTONE_URL       string = "http://127.0.0.1:35357/v3"
+//	KEYSTONE_DOMAIN_ID string = "291d84999cd145cb9dc2eed438021337"
+//)
+
```
Comment (zhuguihua) [inline auth_server/authn/keystone_auth.go]: Please remove unused codes, the same to other places.

File: auth_server/authz/acl_mysql.go [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+	"github.com/golang/glog"
+)
+
+const (
+	KEYSTONE_URL = "http://127.0.0.1:35357/v3"
+	ADMIN_TOKEN  = "123456"
+	DOMAIN_ID    = "61534a660b2a431786cd07233d922d9c"
+	//DOMAIN_ID = "291d84999cd145cb9dc2eed438021337"
+)
+
```
Comment (zhuguihua) [inline auth_server/authz/acl_mysql.go]: Please do not use hard code. These should be parameters to pass by user.

File: auth_server/server/server.go [hunk 5]
```diff
@@ -382,6 +421,220 @@ func (as *AuthServer) Stop() {
...
+		as.doAuthRelDelete(rw, req)
+	default:
+		//default return method not support
+		as.doAuthDefault(rw, req)
+	}
+}
+
+func (as *AuthServer) doAuthRelGet(rw http.ResponseWriter, req *http.Request) {
+	project := req.FormValue("project")
+	if len(project) == 0 {
```
Comment (zhuguihua) [inline auth_server/server/server.go]: Why use len() to judge a string?

File: auth_server/authz/acl_mysql.go (hunk 4 above)
Comment (xwisen) [inline auth_server/authz/acl_mysql.go]: This problem is hard to resolve. If you have better ideas, I will be resolve it .thanks.

File: auth_server/authn/keystone_auth.go [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+// NewKeystoneClient create http client with keystone config
+func NewKeystoneClient(config *KeystoneConfig) (*KeystoneClient, error) {
+	return &KeystoneClient{
+		HttpClient: http.DefaultClient,
+		Config:     config,
+	}, nil
+}
+
+//to keystone api GET /v3/users
+func (kc *KeystoneClient) ListUsers(username string) (r *ListUsersResp, err error) {
```
Comment (zhuguihua) [inline auth_server/authn/keystone_auth.go]: Change function name to GetUser()

**Pull request merged**